labman/gui/handlers/base.py

```diff
@@ -72,3 +74,35 @@ def get(self):
     def head(self):
         self.set_status(404)
         self.finish()
+
+
+class BaseDownloadHandler(BaseHandler):
+    @staticmethod
+    def generate_file_name(name_pieces, process, extension="txt"):
+        date_str = datetime.strftime(process.date,
+                                     process.get_filename_date_format())
+        munged_name_pieces = [re.sub('\s+', '_', x) for x in name_pieces]
+        munged_name_pieces.insert(0, date_str)
+        name_str = "_".join(munged_name_pieces)
+        result = name_str + "." + extension
+        return result
+
+    @authenticated
+    def deliver_text(self, name_pieces, process, text, extension="txt"):
+        output_name = self.generate_file_name(name_pieces, process, extension)
+        self._deliver_file(text, output_name, 'text/csv')
+
+    @authenticated
+    def deliver_zip(self, name_pieces, process, text, extension="zip"):
+        output_name = self.generate_file_name(name_pieces, process, extension)
+        self._deliver_file(text, output_name, 'application/zip')
```

Comment:
I'll leave this up to you but I think the deliver_zip method might be more convenient if it can take the content (whatever it is) and compress it. Alternatively, I believe the parameter named `text` should probably be changed as it is (currently) expected to be a zip archive (I think).

Reply:
You are undoubtedly right about the proposed refactor to deliver_zip, but I don't think I can justify the time to do it before soft launch; I have therefore logged it as an enhancement issue so it doesn't get forgotten (#380  Refactor BaseDownloadHandler.deliver_zip to take in uncompressed content and compress it ).  In the meantime, I have changed the name of the parameter from `text` to `archive`.